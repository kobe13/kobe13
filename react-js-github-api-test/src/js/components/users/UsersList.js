// @flow
import React from 'react';

type UserProps = {
  login: string,
  avatar_url: string,
  id: string,
};

type UsersListProps = {
  users: UserProps[],
  usersNumber: number,
};

const User = ({ login, avatar_url }: UserProps) => (
  <li className="list-group-item text-left">
    <div className="media">
      <img
        alt={`${login} avatar`}
        className="mr-3"
        src={avatar_url}
        width="64"
        height="64"
      />
      <div className="media-body mb-2 mt-2">
        <h5 className="mt-0">{login}</h5>
      </div>
      <a href={`#/user/${login}`} className="btn btn-info">
        See details
      </a>
    </div>
  </li>
);

const UsersList = (props: UsersListProps) => (
  <div className="row">
    <div className="col-12 col-md-6">
      <ul className="list-group">
        {props.users
          .slice(0, props.usersNumber / 2)
          .map(user => <User key={user.id} {...user} />)}
      </ul>
    </div>
    <div className="col-12 col-md-6">
      <ul className="list-group">
        {props.users
          .slice(props.usersNumber / 2, props.usersNumber)
          .map(user => <User key={user.id} {...user} />)}
      </ul>
    </div>
  </div>
);

export default UsersList;
