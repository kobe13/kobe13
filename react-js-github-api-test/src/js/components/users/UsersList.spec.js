import * as React from 'react';
import renderer from 'react-test-renderer';
import UsersList from './UsersList';

test('Users list renders correctly with 2 users', () => {
  const gitHubUsers = [
    {
      login: 'login',
      avatar_url: 'avatar_url',
      key: 1,
    },
    {
      login: 'login2',
      avatar_url: 'avatar_url2',
      key: 2,
    },
  ];
  const component = renderer.create(
    <UsersList users={gitHubUsers} usersNumber={gitHubUsers.length} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
