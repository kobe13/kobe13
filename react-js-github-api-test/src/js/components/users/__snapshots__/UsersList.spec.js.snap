// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users list renders correctly with 2 users 1`] = `
<div
  className="row"
>
  <div
    className="col-12 col-md-6"
  >
    <ul
      className="list-group"
    >
      <li
        className="list-group-item text-left"
      >
        <div
          className="media"
        >
          <img
            className="mr-3"
            src="avatar_url"
            width="64"
          />
          <div
            className="media-body mb-2 mt-2"
          >
            <h5
              className="mt-0"
            >
              login
            </h5>
          </div>
          <a
            className="btn btn-info"
            href="#/user/login"
          >
            See details
          </a>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="col-12 col-md-6"
  >
    <ul
      className="list-group"
    >
      <li
        className="list-group-item text-left"
      >
        <div
          className="media"
        >
          <img
            className="mr-3"
            src="avatar_url2"
            width="64"
          />
          <div
            className="media-body mb-2 mt-2"
          >
            <h5
              className="mt-0"
            >
              login2
            </h5>
          </div>
          <a
            className="btn btn-info"
            href="#/user/login2"
          >
            See details
          </a>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
